src/tracer.c

```diff
@@ -90,7 +88,7 @@ next_finfo(PROCESS_INFO *pi)
 FILE_INFO *
 finfo_at(PROCESS_INFO *pi, int index)
 {
-    if (index == pinfo->finfo_size) {
+    if (index > pinfo->numfinfo) {
```

Comment:
Is this correct? (the case where it's equal, I mean)

Reply:
@zvr isn't `numfinfo` pointing to the last element in the array? In that case if `index` is equal to `pinfo->numfinfo` then the element already exists and thus no need to construct it. Am i missing something?

Reply:
In C, array size 4 means 0,1,2,3.   You definitely cannot do `base+4*size` and expect to get a valid member.

I didn't check what `numfinfo` is now, but at my first write the `_size` variables had the count (e.g. 4).

Reply:
@zvr This should work.